lib/icinga/checkable-check.cpp

```diff
@@ -67,12 +68,19 @@ void Checkable::UpdateNextCheck(const MessageOrigin::Ptr& origin)
 	if (adj != 0.0)
 		adj = std::min(0.5 + fmod(GetSchedulingOffset(), interval * 5) / 100.0, adj);
 
-	double nextCheck = now - adj + interval;
+	double lastCheckScheduleStart = GetLastCheckScheduleStart();
+
+	if (interval > 1)
+		nextCheck = std::max(lastCheckScheduleStart - adj + interval, now + 0.5 + adj);
```

Comment:
Can you please explain the reasoning behind `lastCheckScheduleStart - adj + interval` and `now + 0.5 + adj`? What makes me wonder in particular: `adj` is computed from `now` but you're using it with `lastCheckScheduleStart` and also all other places subtracted `adj` from `now`, here you're adding it.

Reply:
I suspect that `adj` may be designed for this specific statistical distribution in mind and `now` + Random is used only as start number,  There  are two fmod()  methods and if i understand correctly final number will be always less than 5% of `interval`. I'm convinced that if we replace `now` by fixed epoch number ( for example:  1692262263  ) the statistical distribution of `adj` will stay almost the same because of random number .  It was my intention to not change anything that already was tested and proven, thus `adj` calculation was not touched. `now - adj + interval` was simply replaced by `lastCheckScheduleStart - adj + interval` . As for `now + 0.5 + adj` it is arbitrary value in case of server saturation . It will only be used if  `check duration + check latency` will be larger than `interval - adj` .  This is safety precaution, in case plugin executions will not be processed in time. We are adding some delay to execution in that situation instead setting execution time in the past.

It is true that entire  `UpdateNextCheck()` method becoming overcomplicated and probably we could  simplify `adj` calculation.  But there is now way to knowing how new  statistical distribution of `adj` will work for other users. It is not change that i'm competent to propose.